>  failure: The test `ansible-test sanity --test pep8` failed with the following error:
>                           
> test/sanity/pep8/legacy-files.txt:45:1: A201 Remove "lib/ansible/modules/cloud/amazon/elasticache.py" since it passes the current rule set

What am I supposed to do with this error? Is elasticache suddenly legacy? (Especially since the referenced file has not been changed in 5 month)?

--- a/lib/ansible/modules/cloud/amazon/elasticache.py
+++ b/lib/ansible/modules/cloud/amazon/elasticache.py
@@ -208,24 +208,23 @@ class ElastiCacheManager(object):
             else:
                 msg = "'%s' is currently deleting. Cannot create."
                 self.module.fail_json(msg=msg % self.name)
-        if self.cache_port is None:
-            if self.engine == "redis":
-                self.cache_port = 6379
-            elif self.engine == "memcached":
-                self.cache_port = 11211
+
+        kwargs = dict(CacheClusterId=self.name,
+                      NumCacheNodes=self.num_nodes,
+                      CacheNodeType=self.node_type,
+                      Engine=self.engine,
+                      EngineVersion=self.cache_engine_version,
+                      CacheSecurityGroupNames=self.cache_security_groups,
+                      SecurityGroupIds=self.security_group_ids,
+                      CacheParameterGroupName=self.cache_parameter_group,
+                      CacheSubnetGroupName=self.cache_subnet_group,
+                      PreferredAvailabilityZone=self.zone)
+        if self.cache_port is not None:
+            kwargs['Port'] = self.cache_port
 
         try:
-            self.conn.create_cache_cluster(CacheClusterId=self.name,
-                                           NumCacheNodes=self.num_nodes,
-                                           CacheNodeType=self.node_type,
-                                           Engine=self.engine,
-                                           EngineVersion=self.cache_engine_version,
-                                           CacheSecurityGroupNames=self.cache_security_groups,
-                                           SecurityGroupIds=self.security_group_ids,
-                                           CacheParameterGroupName=self.cache_parameter_group,
-                                           CacheSubnetGroupName=self.cache_subnet_group,
-                                           PreferredAvailabilityZone=self.zone,
-                                           Port=self.cache_port)
+            self.conn.create_cache_cluster(**kwargs)
+
         except botocore.exceptions.ClientError as e:
             self.module.fail_json(msg=e.message, exception=format_exc(),
                                   **camel_dict_to_snake_dict(e.response))

--- a/test/sanity/pep8/legacy-files.txt
+++ b/test/sanity/pep8/legacy-files.txt
@@ -43,7 +43,6 @@ lib/ansible/modules/cloud/amazon/ecs_service_facts.py
 lib/ansible/modules/cloud/amazon/ecs_task.py
 lib/ansible/modules/cloud/amazon/ecs_taskdefinition.py
 lib/ansible/modules/cloud/amazon/efs.py
-lib/ansible/modules/cloud/amazon/elasticache.py
 lib/ansible/modules/cloud/amazon/elasticache_subnet_group.py
 lib/ansible/modules/cloud/amazon/elb_instance.py
 lib/ansible/modules/cloud/amazon/elb_classic_lb_facts.py

